Contrib > Toast Component

Blackbaud-SteveBrush opened the pull request:
Original contribution: https://github.com/blackbaud/skyux2/pull/1614

**The following changes were made:**

- Numerous style adjustments that were [decided upon a few days ago](https://github.com/blackbaud/skyux2/issues/1647). (Happy to provide more details around this!)
- Flattened toast folder structure to be more consistent with other modules.
- Moved [component injection logic](https://github.com/blackbaud/skyux2/pull/1676/files#diff-13ba74b7924fed7a3368e770c0440db4R60) from the `SkyToastComponent` to the `SkyToasterComponent`. This was done to make the Toast component behave more like a "normal" component, with inputs, outputs, and transcluded content. The toaster handles both custom and default component injection for the toast's inner content (meaning, the toast component itself now only needs to worry about its core features).
- Renamed the toast service's public method `openTemplatedMessage` to [`openComponent`](https://github.com/blackbaud/skyux2/pull/1676/files#diff-c0167472fc0a6e87ac3f4e916bbc687eR100), since the word "template" has special meaning in Angular, and to further separate the usage of `message` from the `openMessage` method (both methods still operate in the same manner as they did before).
- Created a new class `SkyToast` which holds the configuration aspects of each toast, and slightly modified `SkyToastInstance` to be the publicly consumable "interface" returned from the service methods. (Most of what used to be in `SkyToastInstance` was moved to `SkyToast`.) This will allow us to wrap internal-only utilities in `SkyToast` (for the Toaster component), and only provide what we want the consumer to see in `SkyToastInstance`.

Blackbaud-BobbyEarl commented on src/modules/toast/toaster.component.scss:
```diff
@@ -0,0 +1,17 @@
+  -khtml-user-select: none; /* Konqueror HTML */
+  -moz-user-select: none; /* Firefox */
+  -ms-user-select: none; /* Internet Explorer/Edge */
+  user-select: none; /* Chrom and Opera */
```
Chrome is misspelled.

Blackbaud-BobbyEarl commented on src/modules/toast/toaster.component.scss:
```diff
@@ -0,0 +1,17 @@
+  padding-bottom: $sky-margin-double;
+  padding-right: $sky-margin-double;
+  max-width: 300px;
+  -webkit-touch-callout: none; /* iOS Safari */
```
Is there a reason we wouldn't want users to be able to select text inside a toast?

Blackbaud-SteveBrush commented:
@Blackbaud-BobbyEarl In response to your comments (and for posterity):
- The animation should use "emerge/recede", as described here (I'll implement these changes ASAP): https://developer.blackbaud.com/skyux/design/styles/motion
- I've changed the `string` type to `enum`, as suggested.
- I've removed the hover state, and the `user-select`.
- Let's chat offline about the relationship between toast and toaster.

Blackbaud-SteveBrush commented:
@Blackbaud-BobbyEarl I added an "emerge" animation, as described here: https://developer.blackbaud.com/skyux/design/styles/motion. Also, I added an overflow style to the toaster component to handle a large number of toasts.
